### PR TITLE
[Doc] add-2.3.8-rn

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Signed-off-by: hellolilyliuyi <hellolilyliuyi123@163.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: docs/release_notes/release-2.3.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 # StarRocks version 2.3
 
+## 2.3.8
+
+Release date: February 02, 2023
```
Comment (evelynzhaojie) [inline docs/release_notes/release-2.3.md:5]: 02 改为 2

**Review by amber-create — COMMENTED** [review]
StarRocks performs incremental clone to clone：感觉用词有些重复，建议改为 -> StarRocks incrementally clones

**Review by amber-create — COMMENTED** [review]
When a table enables dynamic partitioning and some partitions are dynamically deleted -》When dynamic partitioning is enabled for a table and some partitions of the table are dynamically deleted

**Review by amber-create — COMMENTED** [review]
returns：语法不对，应该是 is returned；或者主语改一下，StarRocks returns an error xxx.

**Review by amber-create — COMMENTED** [review]
Meanwhile：这里是列举另外一种情况，建议改为 -> Additionally

**Review by amber-create — COMMENTED** [review]
if data load into the table, FE crashes and can not restart -> if you load data into the table, the FEs crash and can not restart

File: docs/release_notes/release-2.3.md [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+- If an FE sends an occasional heartbeat to a BE, and the heartbeat connection times out, the FE considers the BE unavailable, leading to transaction failures on the BE. [# 16386](https://github.com/StarRocks/starrocks/pull/16386)
+- When you use a StarRocks external table to load data between StarRocks clusters, if the source StarRocks cluster is in an earlier version and the target StarRocks cluster is in a later version (2.2.8 ~ 2.2.11, 2.3.4 ~ 2.3.7, 2.4.1 or 2.4.2), the data loading fails. [#16173](https://github.com/StarRocks/starrocks/pull/16173)
+- BEs crash when multiple queries run concurrently and memory usage is relatively high. [#16047](https://github.com/StarRocks/starrocks/pull/16047)
+- When a table enables dynamic partitioning and some partitions are dynamically deleted, if you execute TRUNCATE TABLE, an error NullPointerException returns. Meanwhile, if data load into the table, FE crashes and can not restart. [#16822](https://github.com/StarRocks/starrocks/pull/16822)
+
```
Comment (evelynzhaojie) [inline docs/release_notes/release-2.3.md:17]: When dynamic partitioning is enabled for a table and some partitions are dynamically deleted, if you execute TRUNCATE TABLE, an error NullPointerException is returned. Meanwhile, if data is being loaded into the table, the FE crashes and cannot restart.

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by evelynzhaojie** [comment]
run starrocks_admit_test

**Pull request merged**